shpc/main/registry/__init__.py

```diff
@@ -163,26 +162,39 @@ def sync_from_remote(
         If the registry module is not installed, we install to the first
         filesystem registry found in the list.
         """
-        updates = False
 
+        ## First get a valid local Registry
         # A local (string) path provided
-        if local and isinstance(local, str) and os.path.exists(local):
+        if local and isinstance(local, str):
+            if not os.path.exists(local):
+                logger.exit("The path %s doesn't exist." % local)
             local = Filesystem(local)
 
         # No local registry provided, use default
         if not local:
-            local = Filesystem(self.settings.filesystem_registry)
+            local = self.filesystem_registry
+            # We sync to our first registry - if not filesystem, no go
+            if not local:
+                logger.exit("No local registry to sync to. Check the shpc settings.")
 
-        tmpdir = remote.source
-        if tmpdir.startswith("http") or not os.path.exists(tmpdir):
-            tmpdir = remote.clone()
+        if not isinstance(local, Filesystem):
+            logger.exit("Can only synchronise to a local file system, not to %s." % local)
```

Comment:
Sorry I know US English is terrible - we use z!
```suggestion
            logger.exit("Can only synchronize to a local file system, not to %s." % local)
```